### PR TITLE
Synchronizer L1 parallel: add control between retries to avoid spamming

### LINKED ISSUE #2584: Synchronizer L1 parallel: add control between retries to avoid spamming
# Rationale
If a request to L1 is rejected continously we have to wait a minimum time between retry

# Implementation
Add new config parameter: `MinTimeBetweenRetriesForRollupInfo` to set up the min time between 2 retries for same block range.

### PR DESCRIPTION
**Pull request by joanestebanr**
Closes #2584 
### What does this PR do?
It keep track of the time of the request and if it's too fast it sleep

It add next config parameter: 
```
[Synchronizer.L1ParallelSynchronization]
    MinTimeBetweenRetriesForRollupInfo = "5s"
```
`MinTimeBetweenRetriesForRollupInfo` is the minimum time between retries to request rollup info (it will sleep for fulfill this time) to avoid spamming L1

Main reviewers:

@ARR552

### CODE REVIEW

File: synchronizer/l1_live_block_ranges.go [hunk 1]
```diff
@@ -5,23 +5,26 @@ import (
...
 
-type liveBlockRangeItem struct {
+type liveBlockRangeItem[T any] struct {
 	blockRange blockRange
+	tag        T
 }
 
-type liveBlockRanges struct {
-	ranges []liveBlockRangeItem
+type liveBlockRangesGeneric[T any] struct {
```
Comment (ARR552) [inline synchronizer/l1_live_block_ranges.go:14]: You love to use everything generics

File: synchronizer/l1_live_block_ranges.go [hunk 2]
```diff
@@ -5,23 +5,26 @@ import (
 	"fmt"
 )
 
-type liveBlockRangeItem struct {
+type liveBlockRangeItem[T any] struct {
 	blockRange blockRange
+	tag        T
```
Comment (ARR552) [inline synchronizer/l1_live_block_ranges.go]: What is tag?

File: synchronizer/l1_live_block_ranges.go [hunk 3]
```diff
@@ -59,18 +66,43 @@ func (b *blockRange) overlaps(br blockRange) bool {
...
-func (l *liveBlockRanges) removeBlockRange(br blockRange) error {
+func (l *liveBlockRangesGeneric[T]) setTagByBlockRange(br blockRange, tag T) error {
+	for i, r := range l.ranges {
+		if r.blockRange == br {
+			l.ranges[i].tag = tag
+			return nil
+		}
+	}
+	return errBlockRangeNotFound
+}
```
Comment (ARR552) [inline synchronizer/l1_live_block_ranges.go:94]: Why don't you set the tag in func addBlockRange ?

File: synchronizer/l1_workers_decorator_limit_retries_by_time.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
+package synchronizer
```
Comment (ARR552) [inline synchronizer/l1_workers_decorator_limit_retries_by_time.go:1]: I don't know where to put it but you can take the opportunity to rename the files removing the _ in the names (Snake Case)

File: synchronizer/l1_workers_decorator_limit_retries_by_time.go [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+)
+
+type controlWorkerFlux struct {
+	time    time.Time
+	retries int
+}
+
+type workerDecoratorLimitRetriesByTime struct {
+	*workers
+	processingRanges    liveBlockRangesGeneric[controlWorkerFlux]
```
Comment (ARR552) [inline synchronizer/l1_workers_decorator_limit_retries_by_time.go:24]: I think this controlWorkerFlux shouldn't use liveBlockRanges. This is a completely different structure that is not related with blockRanges.

File: synchronizer/l1_workers_decorator_limit_retries_by_time.go [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+func newWorkerDecoratorLimitRetriesByTime(workers *workers, minTimeBetweenCalls time.Duration) *workerDecoratorLimitRetriesByTime {
+	return &workerDecoratorLimitRetriesByTime{workers: workers, minTimeBetweenCalls: minTimeBetweenCalls}
+}
+
+func (w *workerDecoratorLimitRetriesByTime) String() string {
+	return fmt.Sprintf("[FILTERED_LRBT Active/%s]", w.minTimeBetweenCalls) + w.workers.String()
+}
+
+func (w *workerDecoratorLimitRetriesByTime) asyncRequestRollupInfoByBlockRange(ctx context.Context, blockRange blockRange) (chan responseRollupInfoByBlockRange, error) {
+	ctrl, err := w.processingRanges.getTagByBlockRange(blockRange)
```
Comment (ARR552) [inline synchronizer/l1_workers_decorator_limit_retries_by_time.go:39]: I'm still don't know what is the tag. Could you use a more descriptive name?

**Review by ARR552 — CHANGES_REQUESTED** [review]

File: synchronizer/l1_workers_decorator_limit_retries_by_time.go (hunk 4 above)
Comment (joanestebanr) [reply synchronizer/l1_workers_decorator_limit_retries_by_time.go:1]: the files are snake_case

File: synchronizer/l1_live_block_ranges.go (hunk 1 above)
Comment (joanestebanr) [reply synchronizer/l1_live_block_ranges.go:14]: Yes!!! here you take advantge of this object to be reused

File: synchronizer/l1_live_block_ranges.go (hunk 2 above)
Comment (joanestebanr) [reply synchronizer/l1_live_block_ranges.go]: Tag is a generic field to store any kind of data as extra information, you can use to store related information. 
It's a common field for Microsoft languages as VBA, C#,etc... that allow to add some extra information to already exisiting objects

File: synchronizer/l1_live_block_ranges.go (hunk 3 above)
Comment (joanestebanr) [reply synchronizer/l1_live_block_ranges.go:94]: It's done in that way: first time it call to `addBlockRangeWithTag` that add the range with the tag. 
Later, it's call  `setTagByBlockRange` to update the last time that we try this range

File: synchronizer/l1_workers_decorator_limit_retries_by_time.go (hunk 6 above)
Comment (joanestebanr) [reply synchronizer/l1_workers_decorator_limit_retries_by_time.go:39]: I have add a comment in the definition of the field to clarify the use

File: synchronizer/l1_workers_decorator_limit_retries_by_time.go (hunk 5 above)
Comment (joanestebanr) [reply synchronizer/l1_workers_decorator_limit_retries_by_time.go:24]: That is the reason of using `tag`

**Review by ARR552 — APPROVED** [review]
